packages/autorest.python/autorest/codegen/__init__.py

```diff
@@ -123,7 +123,16 @@ def remove_cloud_errors(yaml_data: Dict[str, Any]) -> None:
     def _build_code_model_options(self) -> Dict[str, Any]:
         """Build en options dict from the user input while running autorest."""
         azure_arm = self.options.get("azure-arm", False)
-        license_header = self.options.get("header-text", DEFAULT_HEADER_TEXT)
+        unbranded = self.options.get("unbranded", False)
```

Comment:
with all of these if-else statements, I think it'd be good to have some kind of client factory in another file that deals with this. Something like

```python
class OptionsRetriever:
    def __init__(self, options: Dict[str, Any], *, azure_arm: bool, unbranded: bool):
        self.options = options
        self.azure_arm = azure_arm
        self.unbranded = unbranded
     @property
     def company_name(self) -> str:
         if self.options.get("company-name"):
           return self.options.get("company-name")
         return "Microsoft" if not self.unbranded else ""
```
Then, do this for all of the options. When initializing the `CodeGenerator` plugin, we also initialize an instance of `OptionsRetriever`. So, now instead of calling self.options.get with all of the unbranded / azure-arm logic, we can just call `self.options_retriever.company_name` etc. This will also help with policy logic and pretty much all of the logic in this file, and the logic itself will be much more neater cleanly written out in another file